### PR TITLE
[docker] fix, Dockerfile for frontend builds

### LINKED ISSUE #9112: Superset-node docker compose fails to create container
Docker-compose off of the master branch fails to launch the superset-node container with `No such file or directory`

### Expected results
Superset-node container starts with docker-compose. 

### Actual results

Superset-node container does not start with docker-compose. 

The docker error is bash: line 0: cd: /app/superset-frontend: No such file or directory
Seems like this commit may have caused the problem:
291306392 (David Aaron Suddjian 2020-02-09 17:53:56 -0800 69)     command: ["bash", "-c", "cd /app/superset-frontend && npm install && npm run dev"]

#### How to reproduce the bug
git clone https://github.com/apache/incubator-superset/
docker-compose up

### Environment
Ubuntu 18.04.3 - AWS

### Checklist

Make sure these boxes are checked before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [ ] I have reproduced the issue with at least the latest released version of superset.
- [ ] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by suddjian**
### CATEGORY

Choose one

- [x] Bug Fix
- [ ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

My previous PR introduced a bug to the Dockerfile. This fixes it by copying build frontend files to/from the right places.

### TEST PLAN

I've verified this by running `docker-compose up --build`, but again would appreciate verification from someone better versed in docker-ology

### ADDITIONAL INFORMATION

- [x] Has associated issue: Fixes #9112
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@mistercrunch @nytai

### CODE REVIEW

**Review by nytai — APPROVED** [review]
👍

**Review by nytai — CHANGES_REQUESTED** [review]
actually getting:
`ERROR: Service 'superset' failed to build: COPY failed: stat /var/lib/docker/overlay2/da84b52e48a6b3377e530754d4fd271917761b201d64bc3a77d4e8574971ba62/merged/app/superset-frontend/version_info.json: no such file or directory` when running locally

**Comment by nytai** [comment]
Looks like `version_info.json` is produced by running setup.py. I think it's only meant to end up in the actual pip tarball package under `superset/static/assets/version_info.json`. Looking at `setup.py` I'm not sure the paths are correct for generating `version_info.json`.

**Comment by suddjian** [comment]
I think you're right, thanks for catching that @nytai

**Comment by suddjian** [comment]
Whew. Had some trials and tribulations as you can see from the commit log but I dare say this is ready to go. Approvals/criticisms welcome.

One change I was forced to make was moving `version_info.json` from `/static/assets` to `/static`, since it is built by python code and webpack cleans the contents of `/assets`. version_info isn't really an asset anyway, so I would argue that `/static` is the correct location for it in this directory structure.

File: docker-compose.yml [hunk 1]
```diff
@@ -26,6 +26,7 @@ x-superset-volumes: &superset-volumes
   - ./docker/docker-init.sh:/app/docker-init.sh
   - ./docker/pythonpath_dev:/app/pythonpath
   - ./superset:/app/superset
+  - ./superset-frontend:/app/superset-frontend
```
Comment (nytai) [inline docker-compose.yml]: 👍

File: Dockerfile [hunk 2]
```diff
@@ -81,6 +82,7 @@ RUN useradd --user-group --no-create-home --no-log-init --shell /bin/bash supers
 COPY --from=superset-py /usr/local/lib/python3.6/site-packages/ /usr/local/lib/python3.6/site-packages/
 # Copying site-packages doesn't move the CLIs, so let's copy them one by one
 COPY --from=superset-py /usr/local/bin/gunicorn /usr/local/bin/celery /usr/local/bin/flask /usr/bin/
+COPY --from=superset-node /app/superset/assets /app/superset/assets
```
Comment (nytai) [inline Dockerfile]: shouldn't these be `/superset/static/assets/` ?

**Comment by nytai** [comment]
should also fix https://github.com/apache/incubator-superset/issues/9108, https://github.com/apache/incubator-superset/issues/9116

File: Dockerfile [hunk 3]
```diff
@@ -81,6 +82,7 @@ RUN useradd --user-group --no-create-home --no-log-init --shell /bin/bash supers
 COPY --from=superset-py /usr/local/lib/python3.6/site-packages/ /usr/local/lib/python3.6/site-packages/
 # Copying site-packages doesn't move the CLIs, so let's copy them one by one
 COPY --from=superset-py /usr/local/bin/gunicorn /usr/local/bin/celery /usr/local/bin/flask /usr/bin/
+COPY --from=superset-node /app/superset/static/assets /app/superset/static/assets
```
Comment (dpgaspar) [inline Dockerfile]: Have not tested it yet, but seems like this should be: 
`COPY --from=superset-node /app/superset/assets /app/superset/static/assets`

**Review by dpgaspar — COMMENTED** [review]
Looks good, needs a possible change

**Review by dpgaspar — APPROVED** [review]
Gave it a quick run and looks good

**Comment by ammarsaf** [comment]
Still happened to me

**Comment by mistercrunch** [comment]
@ammarsaf this issue/PR is super old, would be great if you can find an open issue related to your problem, or start a new one with the details of the branch/command/issue you're having

**Pull request merged**